modules/gcp/storage_test.go

```diff
@@ -0,0 +1,75 @@
+package gcp
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	projectId = os.Getenv("GOOGLE_STORAGE_PROJECT_ID")
+)
+
+func TestCreateAndDestroyStorageBucket(t *testing.T) {
+	t.Parallel()
+
+	id := random.UniqueId()
+	gsBucketName := "gruntwork-terratest-" + strings.ToLower(id)
+	testFilePath := fmt.Sprintf("test-file-%s.txt", random.UniqueId())
+	testFileBody := "test file text"
+
+	logger.Logf(t, "Random values selected Bucket Name = %s, Test Filepath: %s\n", gsBucketName, testFilePath)
+
+	CreateStorageBucket(t, projectId, gsBucketName, nil)
+	defer DeleteStorageBucket(t, gsBucketName)
+
+	// Write a test file to the storage bucket
+	objectURL := WriteBucketObject(t, gsBucketName, testFilePath, strings.NewReader(testFileBody), "text/plain")
+	logger.Logf(t, "Got URL: %s", objectURL)
+
+	// Then verify its contents matches the expected result
+	fileReader := ReadBucketObject(t, gsBucketName, testFilePath)
+
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(fileReader)
+	result := buf.String()
+
+	require.Equal(t, testFileBody, result)
+
+	// Empty the storage bucket so we can delete it
+	EmptyStorageBucket(t, gsBucketName)
```

Comment:
If this test fails, the `defer DeleteStorageBucket(t, gsBucketName)` will run, but it looks like that will only succeed if `EmptyStorageBucket(t, gsBucketName)` runs, so does this need `defer` also?

Reply:
Yes, good point!